OpenRA.Mods.Common/Activities/ExternalCaptureActor.cs

```diff
@@ -38,9 +41,7 @@ public override Activity Tick(Actor self)
 
 			if (IsCanceled || !self.IsInWorld || self.IsDead || !target.IsValidFor(self))
 			{
-				if (capturable.CaptureInProgress)
-					capturable.EndCapture();
-
+				EndCapture(self);
```

Comment:
You need to call this from the case above too, otherwise the condition will remain permanently granted if the target actor dies while being captured.

Reply:
good catch ! Done